game-core/src/main/java/games/strategy/triplea/delegate/MoveDelegate.java

```diff
@@ -189,13 +189,14 @@ public void end() {
       removeAirThatCantLand();
     }
 
-    // WW2V2/WW2V3, fires at end of combat move
-    // WW2V1, fires at end of non combat move
-    if (GameStepPropertiesHelper.isFireRockets(data)) {
-      if (needToDoRockets && TechTracker.hasRocket(bridge.getPlayerId())) {
-        RocketsFireHelper.fireRockets(bridge, bridge.getPlayerId());
-        needToDoRockets = false;
-      }
+    // WW2V1, fires at end of non combat move.
+    // Other versions fire Rockets in BattleDelegate
+    if (needToDoRockets && GameStepPropertiesHelper.isNonCombatMove(data, true)) {
+      if (TechTracker.hasRocket(bridge.getPlayerId())
+          && !Properties.getWW2V2(data) && !Properties.getAllRocketsAttack(data)) {
```

Comment:
Do we also need to check that its not v3? Or do all v3+ maps have getAllRocketsAttack set to true?

Reply:
All the v3 maps that I'm aware of set "All Rockets Attack".

I was going to add referring to the WW2V3 property but then I realised that would propagate the crappy way it was done in the past. So I'm really reluctant to do it that way.

Reply:
Alright. I would think v2 should set that property as well?

Reply:
Just checked, it should but surprisingly it doesn't.

There are 82 map xml's which set WW2V2 without setting "All Rockets Attack". Of these, 50 appear to have a tech step. So we'd need to update a number of maps if we got rid of the WW2V2 check.

Reply:
Ok. We can leave the v2 check then. Just hate that some of these properties are so messy and hard to follow.